src/main/java/org/thoughtcrime/securesms/BaseConversationListFragment.java

```diff
@@ -316,9 +322,37 @@ private void handleSelectAllThreads() {
     updateActionModeItems(actionMode.getMenu());
   }
 
+  private void handleAddToHomeScreen() {
+    final Activity activity = requireActivity();
+    final DcContext dcContext = DcHelper.getContext(activity);
+    final Set<Long> selectedChats = getListAdapter().getBatchSelections();
+    final DcChat chat = dcContext.getChat(selectedChats.iterator().next().intValue());
+
+    Intent intent = new Intent(activity, ShareActivity.class);
+    intent.setAction(Intent.ACTION_SEND);
+    intent.putExtra(ShareActivity.EXTRA_ACC_ID, dcContext.getAccountId());
+    intent.putExtra(ShareActivity.EXTRA_CHAT_ID, chat.getId());
+
+    Recipient recipient = new Recipient(activity, chat);
+    Bitmap avatar = DirectShareUtil.getIconForShortcut(activity, recipient);
+    ShortcutInfoCompat shortcutInfoCompat = new ShortcutInfoCompat.Builder(activity, "chat-" + dcContext.getAccountId() + "-" + chat.getId())
+        .setShortLabel(chat.getName())
+        .setIcon(IconCompat.createWithAdaptiveBitmap(avatar))
+        .setIntent(intent)
+        .build();
+    if (!ShortcutManagerCompat.requestPinShortcut(activity, shortcutInfoCompat, null)) {
+        Toast.makeText(activity, "ErrAddToHomescreen: requestPinShortcut() failed", Toast.LENGTH_LONG).show();
+    } else if (actionMode != null) {
+      actionMode.finish();
+      actionMode = null;
+    }
```

Comment:
was first wondering why a PIN is requested :)

anyways, should't  `actionMode` also be finished in case of errors? not that it makes a big difference here, but this is what we're usually doing. not a blocker or so, just wondering :)

Reply:
if it fails it is likely permission for shortcut was denied and user could go to app settings and enable it again, without needing to select the chat again, which is a relative small gain, selecting the chat again is not a big deal, 

but I think the biggest reason is: if it succeeds, it is finished, but if it fails, it is good that state remains as is to indicate the user that nothing changed and the action didn't succeeded, because otherwise a distracted user might think the shortcut was added to the home screen if they don't read and understand the toast in time and think it was a "added to home scree" toast etc

Reply:
resonabe, thanks for explaining